src/NServiceBus.Core/Hosting/HostingComponent.cs

```diff
@@ -3,41 +3,73 @@
     using System;
     using System.Collections.Generic;
     using System.Diagnostics;
+    using System.Linq;
     using System.Net;
     using System.Runtime;
     using System.Threading.Tasks;
     using Hosting;
-    using Pipeline;
     using Settings;
     using Support;
 
     class HostingComponent
     {
-        HostingComponent(Configuration configuration)
+        HostingComponent(Configuration configuration, List<Type> availableTypes)
```

Comment:
it's interesting that in other places, you pass in the AssemblyScanningComponent to access this information, while here you only pass the types directly. Does that also have an impact on the generated depdendency graph?

Reply:
Fixed